jekyll/_cci2/caching.md

```diff
@@ -1,74 +1,172 @@
+We will cover [dependency caching]() in detail since it where the biggest performance gains are and has more variables and options. [Source caching]() is covered at the end of the document.
 
-### Define Multiple Keys for Cache Restores
+**Note:** On CircleCI 2.0 the Docker images you use for you jobs are automatically cached on our infrastructure where possible. This document is about manually caching dependencies and source code.
 
-Having multiple key candidates for restoring a cache increases the odds of a partial cache hit. Instead of:
+---
+# Caching Dependencies
+ 
+CircleCI 2.0 has more advanced caching strategies available than CircleCI 1.0. We've removed automatic caching to give you finer control. The extra control and power requires that you be explicit about what you cache and how you cache it. You'll learn how to do that in this section.
+
+If you're looking for a quick reference, the [cache configuration options](https://circleci.com/docs/2.0/configuration-reference/#save_cache) are documented in the main Configuration article.
+
+## Minimal `save_cache` example
+
+To save a cache of a file or directory you add the following as a step to a job in you `.circleci/config.yml`:
 
-{% raw %}
-```YAML
-- restore_cache:
-    key: projectname-{{ .Branch }}
 ```
-{% endraw %}
+- save_cache:
+    key: my-cache
+    paths:
+      - my-file.txt
+      - my-project/my-dependencies-directory
+```
+
+The path for directories is relative to the `working_directory` of your job. You can specify an absolute path if you choose.
+
+## What dependencies should I cache?
+
+The dependencies that are good to cache during a job are things like the libraries that your project depends on. For example - the libraries the are installed with `pip` in Python or `npm` for Node.js.
+
+The Docker image(s) you use for your jobs will usually have tools preinstalled that are generic for building projects using the language the image is focused on. For example the `circleci/ruby:2.4.1` image has useful tools like git, openssh-client and gzip preinstalled. So these types of tools that are not explicitly required for your project are good candidates for being stored on the Docker image.
+
+## How do I know what the path is for my dependencies?
+
+The various language dependency managers (e.g. `npm` or `pip`) each have their own paths where dependencies are installed. See our Language guides and demo projects for the specifics for your stack: <https://circleci.com/docs/2.0/demo-apps/>
+
+## Using cache keys and template values to control what is cached and restored
+
+This section is where you will learn about the art of caching. We say this because caching is a balance between getting maximum performance (using a full cache for every build) and reliability (not using an out-of-date or inappropriate cache).
```

Comment:
reliability trade-off is more damaging, e.g. getting unclean/corrupted build; build using stale dependencies